doc/design/cli.md

```diff
@@ -0,0 +1,105 @@
+1. install-client.sh (Only `sqlflow` on a Mac)
+1. install.ps1(Install Docker, `sqlflow`, and the image `sqlflow/sqlflow:latest` on a PC, coming later)
+1. install-client.ps1 (Only `sqlflow` on a PC, coming later)
+
+### Refactoring
+
+Fortunately, there'are only a few places in the REPL codebase that requires SQLFlow core.
+
+#### Workflow
+
+At the moment, SQLFlow workflow calls `repl` to execute a single step, after the refactoring, as `repl` has become a pure client, the workflow code should call the [`step` binary](https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/cmd/step/step.go#L45-L53) instead.
+
+Consequently, we can delete code in `repl` about workflow safely:
+
+1. [`isWorkflowStep`](https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/cmd/repl/repl.go#L314-L320)
+
+1. [Table in protobuf](https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/cmd/repl/repl.go#L161-L163)
+
+1. [The step package](https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/cmd/repl/repl.go#L35)
+
+#### [Command Line Flags](https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/cmd/repl/repl.go#L261-L268)
+
+1. We don't need the flag `--model_dir` any more because the training process is running in `sqlflowserver`.
+
+1. We'll define a new flag `--host` that holds the `sqlflowserver` address.
+
+#### Configuration
+We still use `godotenv` as well as environment variables to hold flags that change infrequently. The two necessary config entries are:
+
+1. `SQLFLOW_DATASOURCE` the alternative of `--datasource`. We already have this.
+
+1. `SQLFLOW_SERVER_HOST` the alternative of `--host`. This is to be added.
+
+#### [RunSQLProgramAndPrintResult](https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/cmd/repl/repl.go#L175)
+ 
+We have to replace the call to `step.RunSQLProgramAndPrintResult` with a new function `RunSQLProgram` that would be defined in the `sqlflow` codebase. The new function should:
+
+1. Construct a client stub from `--host`. 
+
+1. Use the stub to send the SQL statement get from `ReadStmt` to the server.
+
+1. Parse the result returned from the SQLFlow server.
+    - For non-sixel terminal simulators, the server side should return `Plotille` ASCII figures. See https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/pkg/step/step.go#L69-L85
+
+1. Both the client construction code and result parsing code can be implemented by referring to the existing code in `sqlflowserver/main_test.go`.
+
+#### Auto-complete
+The original implementation of auto-complete is based on the `pkg/sql/attribute` package, in a pure client as `sqlflow`, we can only depend on such packages partly because the client may have a different version with the server-side.
```

Comment:
Let us remove this auto-completion feature at the first stage. We need a minimum viable product at the moment. Let us define it clearly in this design doc.

Reply:
Done.